frontend/src/metabase/query_builder/components/AlertListPopoverContent/AlertListPopoverContent.tsx

```diff
@@ -3,46 +3,35 @@ import { useState } from "react";
 import { t } from "ttag";
 import _ from "underscore";
 
-import Modal from "metabase/components/Modal";
 import CS from "metabase/css/core/index.css";
 import { useSelector } from "metabase/lib/redux";
-import { getQuestionAlerts } from "metabase/query_builder/selectors";
 import { getUser } from "metabase/selectors/user";
 import { Icon } from "metabase/ui";
 import type { Alert } from "metabase-types/api";
 
-import { CreateAlertModalContent } from "../AlertModals";
-
 import { AlertListItem } from "./AlertListItem";
 
 type AlertListPopoverContentProps = {
-  setMenuFreeze: (freeze: boolean) => void;
-  closeMenu: () => void;
+  questionAlerts?: Alert[];
+  onCreate: () => void;
+  onEdit: (alert: Alert) => void;
+  onClose: () => void;
 };
 
 export const AlertListPopoverContent = ({
-  setMenuFreeze,
-  closeMenu,
+  questionAlerts,
+  onCreate,
+  onEdit,
+  onClose,
 }: AlertListPopoverContentProps) => {
-  const questionAlerts = useSelector(getQuestionAlerts);
   const user = useSelector(getUser);
 
-  const [adding, setAdding] = useState(false);
   const [hasJustUnsubscribedFromOwnAlert, setHasJustUnsubscribedFromOwnAlert] =
     useState(false);
 
-  const onAdd = () => {
-    setMenuFreeze(true);
-    setAdding(true);
-  };
-
-  const onEndAdding = (shouldCloseMenu = false) => {
-    setMenuFreeze(false);
-    setAdding(false);
-    if (shouldCloseMenu) {
-      closeMenu();
-    }
-  };
```

Comment:
we can simplify a lot of state and get rid of this "Freeze" nonsense by moving the add/edit modals up and out of this popover